src/parser.rs

```diff
@@ -130,10 +130,16 @@ pub enum AstNode {
         block: NodeId,
     },
 
+    /// Long flag ('--' + one or more letters)
+    FlagLong,
+    /// Short flag ('-' + single letter)
+    FlagShort,
+    /// Group of short flags ('-' + more than 1 letters)
+    FlagShortGroup,
```

Comment:
I think we've always supported `-abc` short flag groups in Nushell signatures, not sure what you mean about `extern`. In `extern`, unknown flags are passed as strings. We could split the `FlagShortGroup` into a Vec of `FlagShort` for easier inference of the spance. For example, `-abc` could be parsed as three short flags with spans pointing at `-a`, `b` and `c`?

Reply:
If I recall stuff like the `gcc -Wall` or `^find -name` currently can't be annotated for completion with `extern` as we assume it is always a grouping of shortflags.

Reply:
Yeah, that's true. That would require a syntax change to allow long flags with single dashes, and it would create a potential conflicting state of `def foo [-abc -a -b -c] { ... }`.

We were also talking some time ago about redesigning `extern`s, but I don't recall the details. Not sure if we need to change the signature parsing from current Nushell at this stage, though?

Reply:
Ah that also wouldn't affect the call site flag parsing and only the signature parsing.
I was wondering if we would delay the parsing of the flag groups in the subcomponent but as it is not part of the internal call syntax, we can immediately go with the short flag bunching when we resolve it as an internal.

Reply:
We cannot add new AST nodes after the parsing stage. So, we need to "greedily" parse calls as if they were internal calls, with flags and everything.

Maybe adding these flag nodes was a bit premature, since we don't have the flag parsing in place at all yet. We can evolve it as we go.